File: backend/src/main/java/com/app/backend/dtos/ThamGiaTranDauDTO.java
package com.app.backend.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotNull;
import lombok.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
public class ThamGiaTranDauDTO {
    @NotNull
    @JsonProperty("tran_dau_id")
    private Long tranDauId;

    @JsonProperty("ma_pin")
    private String maPin; // nếu phòng có mã PIN, có thể thêm tuỳ chọn
}

File: frontend/src/app/dtos/tran-dau/thamgiatrandau-dto.ts
export class ThamGiaTranDauDTO {
  tran_dau_id: number;
  ma_pin?: string | null;

  constructor(data: Partial<ThamGiaTranDauDTO>) {
    this.tran_dau_id = data.tran_dau_id ?? 0;
    this.ma_pin = data.ma_pin ?? null;
  }
}
